src/datasets/load.py

```diff
@@ -782,22 +787,21 @@ def __init__(
         download_config: Optional[DownloadConfig] = None,
         download_mode: Optional[DownloadMode] = None,
     ):
-        if data_files is None and data_dir is not None:
-            data_files = os.path.join(data_dir, "**")
 
         self.name = name
         self.data_files = data_files
-        self.downnload_config = download_config
+        self.data_dir = str(Path(data_dir).resolve()) if data_dir else str(Path().resolve())
```

Comment:
(nit) for consistency with the other module factories, I would just leave
```suggestion
        self.data_dir = data_dir
```

and check whether `self.data_dir` is None in `get_module()`